src/js/components/settings/roles.js

```diff
@@ -91,13 +121,16 @@ export class RoleManagement extends React.Component {
                 <TableCell>{role.allowUserManagement ? 'Yes' : 'No'}</TableCell>
                 <TableCell>{role.groups.length ? role.groups.join(', ') : 'All devices'}</TableCell>
                 <TableCell>{role.description || '-'}</TableCell>
-                <TableCell>{role.editable && <Button onClick={() => removeRole(role.id)}>Remove</Button>}</TableCell>
+                <TableCell>
+                  {role.editable && <Button onClick={() => self.editRole(role)}>Edit</Button>}
+                  {role.editable && <Button onClick={() => self.removeRole(role.id)}>Remove</Button>}
+                </TableCell>
               </TableRow>
             ))}
           </TableBody>
         </Table>
 
-        {!adding && <Chip className="margin-top-small" color="primary" icon={<AddIcon />} label="Add a role" onClick={() => self.setState({ adding: true })} />}
+        {!adding && <Chip className="margin-top-small" color="primary" icon={<AddIcon />} label="Add a role" onClick={() => self.addRole()} />}
         <Collapse in={adding} className="margin-right-small filter-wrapper" classes={{ wrapperInner: 'margin-bottom-small margin-right' }}>
```

Comment:
could you combine the two Collapses and just render the internal things conditionally?

Reply:
Done